### PR TITLE
Improve the "missing mods" error screen

### PR DESCRIPTION
**Pull request by mezz**
I realize the log has all the information you need, but I see this error screen pasted all over the place with questions about it, so I figure it can use some improvements to help beginners out and cut down on the reports.

Before:
missing-mods-before

After:
_edit: this is the final result after some discussion below_
missing-mods-after-buttons2

I removed the "Forge Mod Loader blah blah" since it doesn't say anything and people stop reading there.
The message now prominently displays the mod that is missing its dependencies.
It formats the required version strings in a human-readable way (without requiring knowledge of the bracket/braces range notation). It also can display version ranges and maximum versions, which were not previously supported.

### CODE REVIEW

File: src/main/java/net/minecraftforge/fml/client/GuiModsMissing.java [hunk 1]
```diff
@@ -13,7 +13,9 @@
 package net.minecraftforge.fml.client;
 
 import net.minecraft.client.gui.GuiErrorScreen;
+import net.minecraftforge.fml.common.Loader;
 import net.minecraftforge.fml.common.MissingModsException;
+import net.minecraftforge.fml.common.ModContainer;
```
Comment (kashike) [inline src/main/java/net/minecraftforge/fml/client/GuiModsMissing.java]: Unused imports

Comment (mezz) [inline src/main/java/net/minecraftforge/fml/client/GuiModsMissing.java]: thanks

**Comment by simon816** [comment]
I think the use of color would greatly improve the readability of the message. Just highlighting the mod names in a different color will look nicer than plain white text.

**Comment by mezz** [comment]
Interesting idea. How does this look?
missing-mods-after23

**Comment by GirafiStudios** [comment]
I prefer the straight white text to be honest :)

**Comment by simon816** [comment]
Colors need more contrast rather than just shades of white.

**Comment by ChunHin** [comment]
I think you should use "or" instead of "and".
Example:
Mod(1.0 and above)
It is like it requires 1.0 and another version that is above 1.0 (1.1, 1.2 ...).

**Comment by mezz** [comment]
I don't believe adding bright colors adds to the readability, it just makes the error ugly/scary.

**Comment by mezz** [comment]
I think this is a good compromise:

missing-mods-after1231

Also good suggestion @ChunHin

**Comment by darkhax** [comment]
Looks good. This may be a bit outside the scope, but could you make the 'loge/fml-client-latest.log' bit clickable, and have it open the latest log in default editor?

**Comment by Actuarius** [comment]
@mezz added labels [Cleanup]

**Comment by LexManos** [comment]
> but could you make the 'loge/fml-client-latest.log' bit clickable, and have it open the latest log in default editor?
That would be useful.

**Comment by mezz** [comment]
Alright, 
missing-mods-after-buttons

**Comment by KnightMiner** [comment]
This is minor, but maybe underline the clickable parts of the message like the vanilla screenshot links?

**Comment by mezz** [comment]
Only the buttons are clickable.

**Comment by mrAppleXZ** [comment]
Looks great!

**Comment by ProfessorProspector** [comment]
What happens if there's a load of dependencies for whatever reason? Also what if multiple mod's dependencies are missing. For example EnderStorage is missing CodeChickenCore as well. Will it display that as well?

**Comment by mezz** [comment]
Forge aborts as soon as it runs into a mod that's missing dependencies, so it will only show one at a time. Detecting every missing mod is not really possible, fix one issue may require shifting mods versions around, breaking more things... better to keep it simple and stable here.

**Pull request merged**